pkg/clients/zips/client-proxy.go

```diff
@@ -115,19 +137,59 @@ func (c *proxyClient) Get(mod coordinates.Module) (repository.Blob, error) {
 	// request looks like
 	//
 	// GET https://proxy.golang.org/oss.indeed.com/go/taggit/@v/v0.3.3.zip
-	zipURI := c.uriOf(mod)
+	zipURI := c.zipURIOf(mod)
 	c.log.Tracef("making zip proxy request to %s", zipURI)
 
+	response, err := c.sendRequest(mod.String(), zipURI)
+	if err != nil {
+		return nil, err
+	}
+
+	return response, nil
+}
+
+func (c *proxyClient) List(source string) ([]semantic.Tag, error) {
+	// request looks like
+	//
+	// GET https://proxy.golang.org/oss.indeed.com/go/taggit/@v/list
+	listURI := c.listURIOf(source)
+	c.log.Tracef("making list proxy request to %s", listURI)
+
+	response, err := c.sendRequest(source, listURI)
+	if err != nil {
+		return nil, err
+	}
+
+	split := strings.Split(strings.TrimSpace(string(response)), "\n")
```

Comment:
nitpick: it's probably better to use bufio.Scanner with the stdlib line reader, since that uses buffered IO instead of processing the whole user input all at once (could get oomed on accident)

Reply:
I'll change it, although we'll still need to keep all the versions in memory. So this is an improvement for sure, but won't save us if a particular repo has a very large number of versions.